### PR TITLE
PCP-362 use PCP::Client#close rather than EM weirdness

### PR DESCRIPTION
**Pull request by richardc**
ruby-pcp-client 0.3.0 adds a long-missing PCP::Client#close method
to shut down a client after we're done with it.

Here we switch to using this close method with a long-running
eventmachine reactor thread, avoiding the painful races caused by
trying to manage client lifecycle by restarting the supervising
eventmachine.

### CODE REVIEW

**Comment by richardc** [comment]
Can be reviewed now, but will need 0.3.0 of ruby-pcp-client shipped with at least this PR merged:  https://github.com/puppetlabs/ruby-pcp-client/pull/11

File: acceptance/lib/pxp-agent/test_helper.rb [hunk 1]
```diff
@@ -84,10 +84,8 @@ def get_pcp_broker_status(host)
 
 # Start an eventmachine reactor in a thread
```
Comment (james-stocks) [inline acceptance/lib/pxp-agent/test_helper.rb]: This comment might need updated along with the method name being updated

Comment (richardc) [inline acceptance/lib/pxp-agent/test_helper.rb]: How so?  It's still starting an eventmachine reactor in a thread.

File: acceptance/lib/pxp-agent/test_helper.rb [hunk 2]
```diff
@@ -84,10 +84,8 @@ def get_pcp_broker_status(host)
...
 # @return Thread object where the EM reactor should be running
-def start_eventmachine_thread
+def assert_eventmachine_thread_running
```
Comment (james-stocks) [inline acceptance/lib/pxp-agent/test_helper.rb]: The method name may be confusing. But so could the old name.

PCP-298 is a ticket to refactor these helper libs - I'll add a comment specifying that this EventMachine stuff should be contained within a pcp-client helper lib

File: acceptance/lib/pxp-agent/test_helper.rb (hunk 1 above)
Comment (richardc) [inline acceptance/lib/pxp-agent/test_helper.rb]: Could add ", if needed" to be really clear.

File: acceptance/lib/pxp-agent/test_helper.rb (hunk 2 above)
Comment (richardc) [inline acceptance/lib/pxp-agent/test_helper.rb]: OK, how about `ensure_eventmachine_running`?

**Comment by james-stocks** [comment]
Code changes LGTM but I could also try running the tests off your branch before we merge

**Pull request merged**